src/server/worker/api_server.go

```diff
@@ -2121,6 +2130,21 @@ func (a *APIServer) processDatums(pachClient *client.APIClient, logger *taggedLo
 	if err := eg.Wait(); err != nil {
 		return nil, err
 	}
+
+	// put the list of recovered datums in a pfs.Object, and save it as part of the result
+	recoveredDatumsBuf := &bytes.Buffer{}
+	pbw := pbutil.NewWriter(recoveredDatumsBuf)
+	for _, datumHash := range recoveredDatums {
+		if _, err := pbw.WriteBytes([]byte(datumHash)); err != nil {
+			return nil, err
+		}
+	}
+	recoveredDatumsObj, _, err := pachClient.PutObject(recoveredDatumsBuf)
+	if err != nil {
+		return nil, err
+	}
+	result.recoveredDatums = recoveredDatumsObj
```

Comment:
Looking at this a bit more, I think it makes sense to only run this code if `len(recoveredDatums) > 0` and let `result.recoveredDatums` be left `nil` if it's not. And then make the other end that reads it also take nil into account. It doesn't make a lot of sense to write an empty object just to record that there's no recovered datums when `nil` does the job just as well.